src/sonic_ax_impl/mibs/__init__.py

```diff
@@ -34,7 +34,14 @@ def get_index(if_name):
     OIDs are 1-based, interfaces are 0-based, return the 1-based index
     Ethernet N = N + 1
     """
-    match = re.match(SONIC_ETHERNET_RE_PATTERN, if_name.decode())
+    return get_index_from_str(if_name.decode())
+
+def get_index_from_str(if_name):
+    """
+    OIDs are 1-based, interfaces are 0-based, return the 1-based index
+    Ethernet N = N + 1
+    """
+    match = re.match(SONIC_ETHERNET_RE_PATTERN, if_name)
```

Comment:
Let's keep it for readability and optimize if we find it on critical path. re library has the opportunity to hide the optimization without users changing code.

---
In reply to: [108313222](https://github.com/Azure/sonic-snmpagent/pull/19#discussion_r108313222) [](ancestors = 108313222)